filebeat/config/config_test.go

```diff
@@ -104,12 +105,16 @@ func TestMergeConfigFiles(t *testing.T) {
 	assert.NoError(t, err)
 
 	files, err := getConfigFiles(absPath)
-
 	assert.NoError(t, err)
 	assert.Equal(t, 2, len(files))
 
+	var fixedFiles []string
+	for _, f := range files {
+		fixedFiles = append(fixedFiles, testcfg.NewFileWith644Perm(t, f))
+	}
```

Comment:
I'm personally struggling to understand what's going on here. Perhaps we need a comment explaining this?

Reply:
done in https://github.com/elastic/beats/pull/42914/commits/9cfe7de7438024cda539b7c25a965118f41bcece